be/src/formats/parquet/column_reader.cpp

```diff
@@ -578,22 +821,121 @@ class StructColumnReader : public ColumnReader {
+            struct_column = down_cast<StructColumn*>(column.get());
+        }
+        return _child_readers[sub_field]->init_dict_column(struct_column->field_column(sub_field), sub_field_path,
+                                                           layer + 1);
+    }
+
+    void filter_dict_column(const ColumnPtr& column, Filter* filter, const std::vector<std::string>& sub_field_path,
+                            const size_t& layer) override {
+        const std::string& sub_field = sub_field_path[layer];
+        StructColumn* struct_column = nullptr;
+        if (column->is_nullable()) {
+            NullableColumn* nullable_column = down_cast<NullableColumn*>(column.get());
+            DCHECK(nullable_column->mutable_data_column()->is_struct());
+            struct_column = down_cast<StructColumn*>(nullable_column->mutable_data_column());
+        } else {
+            DCHECK(column->is_struct());
+            DCHECK(!_field->is_nullable);
+            struct_column = down_cast<StructColumn*>(column.get());
+        }
+        return _child_readers[sub_field]->filter_dict_column(struct_column->field_column(sub_field), filter,
+                                                             sub_field_path, layer + 1);
+    }
+
+    Status fill_dst_column(ColumnPtr& dst, const ColumnPtr& src) override {
+        StructColumn* struct_column_src = nullptr;
+        StructColumn* struct_column_dst = nullptr;
+        if (src->is_nullable()) {
+            NullableColumn* nullable_column_src = down_cast<NullableColumn*>(src.get());
+            DCHECK(nullable_column_src->mutable_data_column()->is_struct());
+            struct_column_src = down_cast<StructColumn*>(nullable_column_src->mutable_data_column());
+            NullColumn* null_column_src = nullable_column_src->mutable_null_column();
+            NullableColumn* nullable_column_dst = down_cast<NullableColumn*>(dst.get());
+            DCHECK(nullable_column_dst->mutable_data_column()->is_struct());
+            struct_column_dst = down_cast<StructColumn*>(nullable_column_dst->mutable_data_column());
+            NullColumn* null_column_dst = nullable_column_dst->mutable_null_column();
+            null_column_dst->swap_column(*null_column_src);
+            nullable_column_src->update_has_null();
+            nullable_column_dst->update_has_null();
+        } else {
+            DCHECK(src->is_struct());
+            DCHECK(dst->is_struct());
+            DCHECK(!_field->is_nullable);
+            struct_column_src = down_cast<StructColumn*>(src.get());
+            struct_column_dst = down_cast<StructColumn*>(dst.get());
+        }
+        const auto& field_names = struct_column_dst->field_names();
+        for (size_t i = 0; i < field_names.size(); i++) {
+            const auto& field_name = field_names[i];
+            if (LIKELY(_child_readers.find(field_name) != _child_readers.end())) {
+                if (_child_readers[field_name] == nullptr) {
```

Comment:
Why _child_readers[field_name] could be null?

Reply:
struct subfiled may not exist cause of schema change...